File: booking_service.js
const express = require('express');
const cors = require('cors');
const { v4: uuidv4 } = require('uuid');

const app = express();
app.use(cors());
app.use(express.json());

const bookings = {};

app.post('/bookings', (req, res) => {
    const { room_id, check_in, check_out, guest_name, guest_email, total_amount } = req.body;
    
    if (!room_id || !check_in || !check_out || !guest_name || !guest_email) {
        return res.status(400).json({ error: "Missing fields" });
    }
    
    const booking_id = uuidv4().slice(0, 8);
    const booking = {
        booking_id,
        room_id,
        check_in,
        check_out,
        guest_name,
        guest_email,
        status: "PENDING",
        total_amount: total_amount || 0,
        created_at: new Date().toISOString()
    };
    bookings[booking_id] = booking;
    res.status(201).json(booking);
});

app.get('/bookings/:booking_id', (req, res) => {
    const booking = bookings[req.params.booking_id];
    if (!booking) return res.status(404).json({ error: "Booking not found" });
    res.json(booking);
});

app.listen(5002, () => {
    console.log('Booking service running on port 5002');
});

File: booking_service.py
from flask import Flask, request, jsonify
from flask_cors import CORS
import uuid
from datetime import datetime

app = Flask(__name__)
CORS(app)

bookings = {}

@app.route('/bookings', methods=['POST'])
def create_booking():
    data = request.json
    required = ['room_id', 'check_in', 'check_out', 'guest_name', 'guest_email']
    if not all(field in data for field in required):
        return jsonify({"error": "Missing fields"}), 400
    
    booking_id = str(uuid.uuid4())[:8]
    booking = {
        "booking_id": booking_id,
        "room_id": data['room_id'],
        "check_in": data['check_in'],
        "check_out": data['check_out'],
        "guest_name": data['guest_name'],
        "guest_email": data['guest_email'],
        "status": "PENDING",
        "total_amount": data.get('total_amount', 0),
        "created_at": datetime.now().isoformat()
    }
    bookings[booking_id] = booking
    return jsonify(booking), 201

@app.route('/bookings/<booking_id>', methods=['GET'])
def get_booking(booking_id):
    booking = bookings.get(booking_id)
    if not booking:
        return jsonify({"error": "Booking not found"}), 404
    return jsonify(booking)

@app.route('/bookings/<booking_id>/status', methods=['PATCH'])
def update_status(booking_id):
    data = request.json
    new_status = data.get('status')
    if new_status not in ['PENDING', 'CONFIRMED', 'CANCELLED']:
        return jsonify({"error": "Invalid status"}), 400
    if booking_id in bookings:
        bookings[booking_id]['status'] = new_status
        return jsonify(bookings[booking_id])
    return jsonify({"error": "Booking not found"}), 404

if __name__ == '__main__':
    app.run(host='0.0.0.0', port=5002, debug=True)
